hbase-client/src/main/java/org/apache/hadoop/hbase/filter/PrefixFilter.java

```diff
@@ -69,13 +71,19 @@ public boolean filterRowKey(Cell firstRowCell) {
     if ((!isReversed() && cmp > 0) || (isReversed() && cmp < 0)) {
       passedPrefix = true;
     }
-    filterRow = (cmp != 0);
+    filterRow = (cmp > 0);
+    provideHint = cmp < 0;
```

Comment:
Do we maybe need to check if the scan is reversed here? :thinking:

Reply:
Yes, it also needs to be added to the tests.